test/src/engine/BMDieTest.php

```diff
@@ -141,6 +141,128 @@ public function testCreate_from_recipe() {
         $this->assertTrue($die->has_skill('Poison'));
         $this->assertTrue($die->has_skill('Shadow'));
         $this->assertEquals(6, $die->max);
+
+        try {
+            $die = BMDie::create_from_recipe('pp(5)', TRUE);
+            $this->fail('Creating die with doubled skills did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('{p,s}(4)', TRUE);
+            $this->fail('Creating die with plasma skill did not throw an exception.');
+        } catch (BMExceptionUnimplementedSkill $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(2000)', TRUE);
+            $this->fail('Creating exceedingly large die did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(-1)', TRUE);
+            $this->fail('Creating negative valued die did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(9.0)', TRUE);
+            $this->fail('Creating decimal die did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(9.1)', TRUE);
+            $this->fail('Creating non-integer die did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(X,Y)', TRUE);
+            $this->fail('Creating non-identical twin swing die did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(p6)', TRUE);
+            $this->fail('Creating invalid die type did not throw an exception.');
+        } catch (BMExceptionDieRecipe $e) {
+
+        }
+
+        try {
+            $die = BMDie::create_from_recipe('(6) +(4) +(2)', TRUE);
```

Comment:
Note to self: this test doesn't make sense here, it should probably be in BMButtonTest